<a href="https://colab.research.google.com/github/ZozolNat/teor_veroiatnosti-i-stat/blob/main/%D0%B43_6_%D1%81%D1%82%D0%B0%D1%82_%D0%B8_%D0%B2%D0%B5%D1%80%D0%BE%D1%8F%D1%82_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

Описание решения:
Т.к нам изветно среднее квадрат. отклонение(σ), можно использовать формулу для доверительного интервала при известном нормальным(стандартным) отклонении. Доверительный интервал = [M - Z * (σ/√n), M + Z * (σ/√n)]

где:

M = 80 (выборочное среднее)

Z = (значение стандартного нормального распределения для выбранной надежности)

σ = 16 (стандартное отклонение генеральной совокупности)

n =256 (объем выборки)

0,95 Напдежность

In [6]:


import scipy.stats as stats
# иисходное дано
M = 80
σ = 16
n = 256
safety = 0.95

# ищем значение Z для выбранной надежности (у нас надежность 0,95)
Z = stats.norm.ppf((1 + safety) / 2)
print("значение Z для выбранной надежности:", Z)

## stats.norm.ppf функциz из библиотеки SciPy,  используется для вычисления обратного значения (квантиля) стандартного нормального распределения. Функция ppf (Percent Point Function) принимает в качестве аргумента вероятность и возвращает значение Z-статистики, соответствующее этой вероятности. Для того чтоб здесь найти доверительный интервал, используется stats.norm.ppf для определения значения Z, которое соответствует выбранной надежности. Значение Z используется для определения ширины доверительного интервала вокруг выборочного среднего.


значение Z для выбранной надежности: 1.959963984540054


In [3]:
# Вычисляем доверительный интервал
lower_bound = M - Z * (σ / (n ** 0.5))
upper_bound = M + Z * (σ / (n ** 0.5))

print("Доверительный интервал:", (lower_bound, upper_bound))

Доверительный интервал: (78.04003601545995, 81.95996398454005)


#2 В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

описание  и решение. Т.к. точное значение стандартного отклонения генеральной совокупности неизвестно, а объем выборки небольшой (n < 30), то для оценки истинного значения величины X с использованием доверительного интервала, можно воспользоваться t-распределением.

1 ) сначла вычислим выборочное среднее (M) и выборочное стандартное отклонение (S) по данным измерений.

M = sum(X) / len(X) = (6.9 + 6.1 + 6.2 + 6.8 + 7.5 + 6.3 + 6.4 + 6.9 + 6.7 + 6.1) / 10 = 6.59

S = sqrt(sum((x - M)**2) / (len(X) - 1)) = sqrt(((6.9 - 6.59)**2 + (6.1 - 6.59)**2 + (6.2 - 6.59)**2 + (6.8 - 6.59)**2 + (7.5 - 6.59)**2 + (6.3 - 6.59)**2 + (6.4 - 6.59)**2 + (6.9 - 6.59)**2 + (6.7 - 6.59)**2 + (6.1 - 6.59)**2) / 9) = 0.478

2)двлее надо найти значен. t для выбранной доверительной вероятности и степеней свободы. Степень свободы считается = (n - 1), в данном случае n=10, значит получаетя 9. и таким образом принимаем, что надо найти t lля 95% доверительного интервала и 9 степеней свободы.

In [7]:
import scipy.stats as stats
#указываем что дано
safety = 0.95
degreefree = 9

t = stats.t.ppf((1 + safety) / 2, degreefree)
print("Значение t:", t)


Значение t: 2.2621571627409915


3) дажее ищем стандартную ошибку среднего (SE), используя формулу SE = S / sqrt(n), где S - выборочное стандартное отклонение ( 0,478 нашли в пункте один), n - объем выборки(в нашем случае 10). SE = S / sqrt(n) = 0.478 / sqrt(10) = 0.151

*ниже приведено в коде

In [19]:
from numpy import sqrt
import scipy.stats as stats
#указываем что дано
S = 0,478
n = 10

# Вычисление стандартной ошибки среднего (SE)
SE = S/sqrt(n)
print("вычисление стандатной ошибки среднего:", SE)

вычисление стандатной ошибки среднего: [  0.         151.15687216]


4) далее надо вычислить предел погрешности (E), для этого используем формулу E = t * SE. получается E = 2.262*0.151=0.3415

5) чтоб определить нижнюю и верхнюю границы доверительного интервала, надо вычесть и прибавить предел погрешности к выборочному среднему: Нижняя граница = M - E, Верхняя граница = M + E., т.о. Нижняя граница = 6,59-0,3415=6,2485 и Верхняя граница =6,59+0,3415=6,9315

ответ: доверительный интервал для оценки истинного значения величины X с доверительной вероятностью 0.95 составляет от 6.2485 до 6.9315.

#3)Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170. Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175. Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.


Описание и решение. чтоб построить 95% доверительный интервал для разности среднего роста родителей и детей, надо сначала вычислить средние значения роста дочерей и матерей, потом расчет стандартного отклрнения по матерям и дочепям, далее рассчет стандартной ошибки и вычисление интервала. 
1) cчитаем  средний рост дочерей:

Средний рост дочерей: mean_daughters = sum(daughters) / len(daughters)=175+167+154+174+178+148+160+167+169+170/10= 166,2 (складываем все данные по росту дочерей и делим на количество этих данных , в нашем случае 10)

считаем средний рост матерей:
Средний рост матерей: mean_mothers = sum(mothers) / len(mothers) = ( считаем аналогично)= 169.1

2)Вычисляем стандартное отклонение роста дочерей и матерей

*чтоб вычислить стандартное отклонение надо: Вычислите среднее значение (среднее арифметическое) ряда данных, что мы уже сделали в пункте один, далее вычислите разницу между каждым значением и средним значением, т.е. по всем значение например дочерей пройтись и вычесть из каждого значения среднее значение, пото возвести каждую разницу в квадрат (чтобы убрать отрицательные значения и подчеркнуть большие отклонения), а затем вычислить среднее значение полученных квадратов разностей. Извлечь квадратный корень из среднего значения квадратов разностей. Это и будет стандартное отклонение. Мы подробно это делали и расписывади в предыдущих домашниз задачах (см дз 4).

**в пайтоне можно использовать функцию numpy.std() из библиотеки NumPy.(которая вычисляет стандартное отклонение), тогда это  будет выглядеть так в коде:



In [21]:
import numpy as np

# Данные о росте дочерей
daughters = [175, 167, 154, 174, 178, 148, 160, 167, 169, 170]

# Вычисление стандартного отклонения
std_daughters = np.std(daughters)
print("вычисление стандатного отклонение по дочерям:", std_daughters)

вычисление стандатного отклонение по дочерям: 9.05317623820502


In [22]:
import numpy as np

# Данные о росте матерей
mothers = [178, 165, 165, 173, 168, 155, 160, 164, 178, 175]

# Вычисление стандартного отклонения
std_mothers = np.std(mothers)
print("вычисление стандатного отклонение по метерям:", std_mothers)

вычисление стандатного отклонение по метерям: 7.354590403278759


3)делаем рассчет стандартной ошибки: SE=np.sqrt((std_daughters**2) / len(daughters) + (std_mothers**2) / len(mothers))

In [27]:
import numpy as np

# дано
len(daughters) == 10
len(mothers) == 10

# Вычисление стандартного отклонения
std_daughters = 9.05317623820502
std_mothers = 7.354590403278759

# Вычисление стандартной ошибки
se = np.sqrt((std_daughters**2) / len(daughters) + (std_mothers**2) / len(mothers))

print("Стандартная ошибка:", se)


Стандартная ошибка: 3.6884956283015984


4) ищем доверительный интервал.

A) сначала надо найти значение t-статистики для заданной доверительной вероятности и степеней свободы(количество наблюдений - 1. В нашем случаем 10 наблюдений по матерям минус 1 и 10 наблюдений по дочепям минус 1 , т.е 18 степень свободы), а доверительная вероятность 0,975 =( считаем также, как и в задаче выше  (1 + доверительная вероятность/safety = 0.95) / 2), так как доверительный интервал 95%  =0.95).
, таким образом степень свободы 18, доверительная вероятность 18

t = stats.t.ppf(0.975, 18)

In [28]:
import scipy.stats as stats

t = stats.t.ppf(0.975, 18)
print("t", t)

t 2.10092204024096


B) считаем доверительный интервал по формулам , указанным в предыдущей задачи:

Нижняя граница: lower = (mean_daughters - mean_mothers) - (t * se)
Верхняя граница: upper = (mean_daughters - mean_mothers) + (t * se)


In [40]:
import scipy.stats as stats
# из того что нам уже дано , т.е нашли ранее 
mean_daughters = 166.2
mean_mothers = 169.1
t = 2.1009
se = 3.6884
l = (mean_daughters - mean_mothers) - (t * se)
up = (mean_daughters - mean_mothers) + (t * se)

confidence_interval = (l, up)
print("95% доверительный интервал для разности среднего роста родителей и дочерей:", confidence_interval)

95% доверительный интервал для разности среднего роста родителей и дочерей: (-10.648959560000007, 4.8489595599999955)
